src/vstart.sh

```diff
@@ -877,10 +883,23 @@ EOF
             # designate a single CPU node $osd for osd.$osd
             extra_seastar_args="--smp 1 --cpuset $osd"
         fi
+        local osd_pid
         run 'osd' $osd $SUDO $CEPH_BIN/$ceph_osd \
             $extra_seastar_args $extra_osd_args \
-            -i $osd $ARGS $COSD_ARGS
+            -i $osd $ARGS $COSD_ARGS &
+        osd_pid=$!
+        if $parallel; then
+            osds_wait=$osd_pid
+        else
+            wait $osd_pid
+        fi
     done
+    if $parallel; then
+        for p in $osds_wait; do
+            wait $p
+        done
```

Comment:
you can simply use `wait` and avoid recording the `osd_pid`s and this loop.

Reply:
@aclamk ping?

Reply:
@batrick We must have loop. `wait` with single pid copies return value from process. `wait` without pid waits for everyting to complete and succeds.

Reply:
> `wait` without pid waits for everyting to complete and succeds.

Isn't that what you want to do here?

Reply:
No, when I run multiple processes in background, and some of them have non-zero return, then `wait` is ok with this, while with `wait <pid>` I got error.

Reply:
got it